library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java

```diff
@@ -150,25 +151,73 @@ void shouldNotAddOrGetIllegalProperties() {
         assertThat(vertex.property("THIS_DOES_NOT_EXIST")).isEqualTo(VertexProperty.empty());
     }
 
+    @Test
+    void shouldOnlyCreateValidGafferPopVertexPropertyObjects() {
+        // Given
+        final GafferPopGraph graph = mock(GafferPopGraph.class);
+        given(graph.features()).willReturn(features);
+        given(features.vertex()).willReturn(vertexFeatures);
+        given(vertexFeatures.properties()).willReturn(vertexPropertyFeatures);
+        final String propValue = "propValue";
+        final String nestedKey = "nestedKey";
+        final String nestedValue = "nestedValue";
+        // Make new vertex with the mocked bits
```

Comment:
Not sure these comments are required. Seems self explanatory to me.

Reply:
True but doesn't harm to have comments though, doesn't make it any harder to follow